x/gov/client/rest/rest.go

```diff
@@ -199,46 +204,121 @@ func queryProposalHandlerFn(storeName string, cdc *wire.Codec, kb keys.Keybase,
 	}
 }
 
+func queryDepositHandlerFn(storeName string, cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		strProposalID := vars[RestProposalID]
+		bechDepositerAddr := vars[RestDepositer]
+
+		if len(strProposalID) == 0 {
+			w.WriteHeader(http.StatusBadRequest)
+			err := errors.New("proposalId required but not specified")
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		proposalID, err := strconv.ParseInt(strProposalID, 10, 64)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			err := errors.Errorf("proposalID [%d] is not positive", proposalID)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		if len(bechDepositerAddr) == 0 {
+			w.WriteHeader(http.StatusBadRequest)
+			err := errors.New("depositer address required but not specified")
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		depositerAddr, err := sdk.GetAccAddressBech32(bechDepositerAddr)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			err := errors.Errorf("'%s' needs to be bech32 encoded", RestDepositer)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		ctx := context.NewCoreContextFromViper()
+
+		res, err := ctx.QueryStore(gov.KeyDeposit(proposalID, depositerAddr), storeName)
+		if err != nil || len(res) == 0 {
+			res, err := ctx.QueryStore(gov.KeyProposal(proposalID), storeName)
+			if err != nil || len(res) == 0 {
+				w.WriteHeader(http.StatusNotFound)
+				err := errors.Errorf("proposalID [%d] does not exist", proposalID)
+				w.Write([]byte(err.Error()))
+				return
+			}
+			w.WriteHeader(http.StatusNotFound)
+			err = errors.Errorf("depositer [%s] did not deposit on proposalID [%d]", bechDepositerAddr, proposalID)
+			w.Write([]byte(err.Error()))
+			return
+		}
+
+		var deposit gov.Deposit
+		cdc.MustUnmarshalBinary(res, &deposit)
+		depositRest := gov.DepositToRest(deposit)
+		output, err := wire.MarshalJSONIndent(cdc, depositRest)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
+			return
+		}
+		w.Write(output)
+	}
+}
+
 func queryVoteHandlerFn(storeName string, cdc *wire.Codec, kb keys.Keybase, ctx context.CoreContext) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		vars := mux.Vars(r)
-		strProposalID := vars[ProposalRestID]
+		strProposalID := vars[RestProposalID]
 		bechVoterAddr := vars[RestVoter]
 
 		if len(strProposalID) == 0 {
-			w.WriteHeader(http.StatusBadRequest)
+			// w.WriteHeader(http.StatusBadRequest)
```

Comment:
Do we not need these headers?

Reply:
Oh oops, commented these out for some debugging thing.  Will uncomment them